src/components/FileUploadModal/index.tsx

```diff
@@ -0,0 +1,117 @@
+import { Button, Form, Tabs } from "antd";
+import { RcFile } from "antd/lib/upload";
+import React, { useState } from "react";
+import { ActionCreator } from "redux";
+
+import {
+    ClearSimFileDataAction,
+    RequestLocalFileAction,
+} from "../../state/trajectory/types";
+import {
+    SetErrorAction,
+    SetViewerStatusAction,
+} from "../../state/viewer/types";
+
+import CustomModal from "../CustomModal";
+import UrlUploadForm from "./url-upload-form";
+import LocalFileUpload from "../LocalFileUpload";
+import uploadFiles from "./upload-local-files";
+
+import styles from "./style.css";
+
+interface FileUploadModalProps {
+    setIsModalVisible: (isModalVisible: boolean) => void;
+    loadLocalFile: ActionCreator<RequestLocalFileAction>;
+    setViewerStatus: ActionCreator<SetViewerStatusAction>;
+    clearSimulariumFile: ActionCreator<ClearSimFileDataAction>;
+    setError: ActionCreator<SetErrorAction>;
+}
+
+const FileUploadModal: React.FC<FileUploadModalProps> = ({
+    setIsModalVisible,
+    clearSimulariumFile,
+    loadLocalFile,
+    setViewerStatus,
+    setError,
+}) => {
+    const [openTab, setOpenTab] = useState("device");
+    const [noUrlInput, setNoUrlInput] = useState(true);
+    const [fileList, setFileList] = useState<RcFile[]>([]);
+    const [urlForm] = Form.useForm();
+
+    const closeModal = () => {
+        setIsModalVisible(false);
+    };
+
+    const onUrlInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setNoUrlInput(event.target.value.length === 0);
+    };
+
+    const disableLoad =
+        openTab === "device" ? fileList.length === 0 : noUrlInput;
+
```

Comment:
this was immediately obvious to me, ie why when we're loading a local file we'd disable the loading when the file list is zero? Can you add a comment here to explain the logic?

Reply:
also, to make sure you're not getting out of sync, shouldn't you be accessing the openTab through `useEffect`?

Reply:
I believe the body of the render function is a safe place to access state, it's just handlers where that state has been changed that can give inconsistent values.